tests/e2e/kcs/test_selinux_relabel_solution.py

```diff
@@ -0,0 +1,247 @@
+import logging
+import pytest
+import yaml
+import random
+import time
+import tempfile
+
+from ocs_ci.ocs import ocp, constants
+from ocs_ci.helpers import helpers
+from ocs_ci.ocs.exceptions import PodNotCreated
+from ocs_ci.ocs.resources import pod as res_pod
+from ocs_ci.ocs.resources.pod import (
+    wait_for_pods_to_be_running,
+)
+from ocs_ci.framework.testlib import E2ETest
+from ocs_ci.utility.utils import run_cmd
+from ocs_ci.utility.templating import dump_data_to_temp_yaml
+from ocs_ci.framework import config
+
+log = logging.getLogger(__name__)
+
+
+class TestSelinuxrelabel(E2ETest):
+    def create_deploymentconfig_pod(self, **kwargs):
+        """
+        create deployment pod.
+        """
+
+        # Create service_account to get privilege for deployment pods
+        self.sa_name = helpers.create_serviceaccount(namespace=self.project_namespace)
+
+        helpers.add_scc_policy(
+            sa_name=self.sa_name.name,
+            namespace=constants.OPENSHIFT_STORAGE_NAMESPACE,
+        )
+        try:
+            pod_obj = helpers.create_pod(
+                interface_type=constants.CEPHFS_INTERFACE,
+                pvc_name=self.pvc_obj.name,
+                namespace=self.project_namespace,
+                sa_name=self.sa_name.name,
+                dc_deployment=True,
+                pod_dict_path=constants.FEDORA_DC_LINUXTAR_FILES_YAML,
+                **kwargs,
+            )
+        except Exception as e:
+            log.exception(
+                f"Pod attached to PVC {pod_obj.name} was not created, exception [{str(e)}]"
+            )
+            raise PodNotCreated("Pod attached to PVC was not created.")
+        return pod_obj
+
+    def data_integrity_check(self, pod_obj, namespace):
+        """
+        Check data integrity on pod.
+        """
+        pod_name = pod_obj.name
+        log.info(f"Pod name is -------- {pod_name}")
+        ocp_obj = ocp.OCP(
+            kind=constants.POD,
+            namespace=namespace,
+        )
+        data_path = f"{constants.FLEXY_MNT_CONTAINER_DIR}"
+        num_of_files = random.randint(1, 9)
+        random_file = ocp_obj.exec_oc_cmd(
+            f"exec -it {pod_name} -- /bin/bash"
+            f' -c "find {data_path} -type f | "shuf" -n {num_of_files}"',
+            timeout=300,
+        )
+        log.info(f"files are {random_file}")
+        ini_md5sum_pod_data = res_pod.cal_md5sum(pod_obj=pod_obj, file_name=random_file)
+        pod_objs = res_pod.get_all_pods(
+            namespace=namespace,
+            selector=[self.pod_selector],
+            selector_label="deploymentconfig",
+        )
+
+        # Respin pod
```

Comment:
as discussed you dont need to do pod restarts in the data_integrity function, please check